src/main/java/lotto/service/LottoService.java

```diff
@@ -0,0 +1,51 @@
+package lotto.service;
+
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import lotto.Lotto;
+import lotto.LottoNumberGenerator;
+import lotto.LottoRank;
+import lotto.WinningLotto;
+import lotto.LottoResult;
+
+public class LottoService {
+    private final LottoNumberGenerator lottoNumberGenerator;
+
+    public LottoService(LottoNumberGenerator lottoNumberGenerator) {
+        this.lottoNumberGenerator = lottoNumberGenerator;
+    }
+
+    public List<Lotto> purchaseLottos(int purchaseAmount) {
+        int lottoCount = getLottoCount(purchaseAmount);
+
+        List<Lotto> lottos = new ArrayList<>();
+        for (int i = 0; i < lottoCount; i++) {
+            lottos.add(new Lotto(lottoNumberGenerator.generate()));
+        }
+        return lottos;
+    }
+
+    public int getLottoCount(int purchaseAmount) {
+        if (purchaseAmount % 1000 != 0) {
+            throw new IllegalArgumentException("[ERROR] 로또 구입 금액은 1,000원 단위여야 합니다.");
+        }
+        return purchaseAmount / 1000;
+    }
+
+    public LottoResult calculateResults(List<Lotto> lottos, WinningLotto winningLotto) {
+        Map<LottoRank, Integer> rankCounts = new EnumMap(LottoRank.class);
+
+        for (Lotto lotto : lottos) {
+            Optional<LottoRank> optionalRank = winningLotto.match(lotto);
+
+            optionalRank.ifPresent(rank -> {
+                rankCounts.put(rank, rankCounts.getOrDefault(rank, 0) + 1);
```

Comment:
merge() 메서드를 활용하는 것을 추천드립니다! 더 가독성 있고 간결하게 작성할 수 있습니다.
```suggestion
                rankCounts.merge(rank, 1, Integer::sum);
```